state/pool.go

```diff
@@ -8,18 +8,49 @@ import (
 	"fmt"
 	"runtime/debug"
 	"sync"
+	"time"
 
 	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/pubsub"
 	"gopkg.in/juju/names.v2"
+	"gopkg.in/juju/worker.v1"
+
+	"github.com/juju/juju/state/watcher"
 )
 
+var errPoolClosed = errors.New("pool closed")
+
 // NewStatePool returns a new StatePool instance. It takes a State
 // connected to the system (controller model).
 func NewStatePool(systemState *State) *StatePool {
-	return &StatePool{
+	pool := &StatePool{
 		systemState: systemState,
 		pool:        make(map[string]*PoolItem),
+		hub:         pubsub.NewSimpleHub(nil),
+	}
+	// If systemState is nil, this is clearly a test, and a poorly
+	// isolated one. However now is not the time to fix all those broken
+	// tests.
+	if systemState != nil {
```

Comment:
can we invert this to avoid indentation and jeep the code closer to what it should be without the work around
if systemState == nil {
  return pool
}

Reply:
ok